### DISCUSSION TITLE
Automatic z offset calibration

### DISCUSSION THREAD

**Discussion by Otann** [kevinakasam/KlackEnder-Probe#29] [Ideas]
I've recently had a really wild idea...
What if 

- we take a retainer model
- fuse it together with the [z endstop](https://github.com/VoronDesign/Voron-2/blob/Voron2.4/STLs/Z_Endstop/nozzle_probe.stl) from Voron (or somehow reuse the original ender'a z endstop pcb) 
- and use it to run [klipper z calibration](https://github.com/protoloft/klipper_z_calibration) script? 

I think in theory this would allow us to never ever touch the z offset again?

**Comment by Otann** [comment]
I thought about it a bit more, the calibration script requres to touch the end-stop with both nozzle and the probe, but is it even possible with klack?

**Comment by DMNerd** [comment]
It would probably require either two switches for the endstop, to be able to touch both the probe and the nozzle, or having the mount without the probe be flush with the nozzle (whcich would cause deviations). 

Two seitch endstop could be designed pretty easily IMHO
